core/vm/evm.go

```diff
@@ -389,7 +389,7 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	// Create a new account on the state
 	snapshot := evm.StateDB.Snapshot()
 	evm.StateDB.CreateAccount(address)
-	if evm.ChainConfig().IsEIP158(evm.BlockNumber) {
```

Comment:
I notice there are still a few places use `evm.ChainConfig()` instead of chainRule like L47, L201, L204, L414, L431. Please change them.
Just a thought, could we offer a public method called `evm.ChainRule()` so that outside of `core/vm` package we can also use chain rule. So that we can totally get rid of chain config.

Reply:
I'm touching some of these with #19904, and am happy to update these to use `evm.chainRule` over there if that helps. Also happy to wait for that to happen here and then rebase my work, of course.

Reply:
Fixed, ptal @rjl493456442